app/controllers/MetricsController.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright 2016 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package controllers;
+
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.JmxReporter;
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.Timer;
+import com.codahale.metrics.health.HealthCheckRegistry;
+import com.codahale.metrics.health.jvm.ThreadDeadlockHealthCheck;
+import com.codahale.metrics.jvm.MemoryUsageGaugeSet;
+import com.linkedin.drelephant.analysis.AnalyticJob;
+import com.linkedin.drelephant.metrics.CustomGarbageCollectorMetricSet;
+import org.apache.log4j.Logger;
+import play.Configuration;
+import play.libs.Json;
+import play.mvc.Controller;
+import play.mvc.Result;
+
+import static com.codahale.metrics.MetricRegistry.name;
+
+
+/**
+ * This class enables the use of <a href="http://metrics.dropwizard.io">Dropwizard</a>
+ * metrics for the application.
+ *
+ * <br><br>
+ * The following endpoints are exposed.
+ * <ul>/ping - Reports application status if up</ul>
+ * <ul>/healthcheck - Returns status in Json format from all the implemented healthchecks</ul>
+ * <ul>/metrics - Returns all the metrics in Json format</ul>
+ */
+public class MetricsController extends Controller {
+  private static final Logger LOGGER = Logger.getLogger(MetricsController.class);
+
+  private static MetricRegistry METRIC_REGISTRY = null;
+  private static HealthCheckRegistry HEALTH_CHECK_REGISTRY = null;
```

Comment:
Nit: the convention in this project has been to use all caps for `static final` variables only. Since these aren't final they should just be standard camel case.
If the reason you made them all uppercase is because you'd like them to be final but need to check the feature flag before initializing then you should make them final and use a static initializer.